87.md

```diff
@@ -0,0 +1,136 @@
+NIP-87
+======
+
+Ecash Mint Discoverability
+--------------------------------
+
+`draft` `optional`
+
+This NIP describes `kind:38173`, `kind:38172` and `kind:38000`: a way to discover ecash mints, their capabilities, and people who recommend them.
+
+## Rationale
+
+Nostr's discoverability and transparent event interaction is one of its most interesting/novel mechanics.
+This NIP provides a simple way for users to discover ecash mints recommended by other users and to interact with them.
+
+### Parties involved
+
+There are three actors to this workflow:
+
+* An ecash mint operator, announces their mint and its capabilities.
+    * Publishes `kind:38173` or `kind:38172`, detailing how to connect to it and its capabilities.
+* user A, who recommends an ecash mint
+    * Publishes `kind:38000`
+* user B, who seeks a recommendation for an ecash mint
+    * Queries for `kind:38000` and, based on results, queries for `kind:38173`/`kind:38172`
+
+## Events
+
+### Recommendation event
+```json
+{
+  "kind": 38000,
+  "pubkey": <recommender-user-pubkey>,
+  "tags": [
+    ["k", "38173"],
+    ["d", "<d-identifier>"],
+    ["u", <recommended-fedimint-invite-code>],
+    ["a", "38173:fedimint-pubkey:<d-identifier>", "wss://relay1"]
+  ],
+  "content": "I trust this mint with my life"
+}
+```
+
+The recommendation event is a parameterized-replacable event so that a user can change edit their recommendation without creating a new event.
+
+The `d` tag in `kind:38000` is the `kind:38173`/`kind:38172` event identifier this event is recommending, if no event exists, the `d` tag can still be calculated from the mint's pubkey/id.
```

Comment:
What is expected if the mint does not share their pubkey? Lots of the Cashu mints I see do not have a pubkey at their /info endpoint

Reply:
That's the deprecated spec, should be on /v1/info

Reply:
Yea that too, I think it's mostly lnbits mints. 
ie. https://legend.lnbits.com/cashu/api/v1/Na859jHdYk9EjwSiBXurtp/v1/info does not return a pubkey.
I've just been falling back to random d tag

Reply:
cc @callebtc maybe be a bug here?

Reply:
> What is expected if the mint does not share their pubkey? Lots of the Cashu mints I see do not have a pubkey at their /info endpoint

LNbits mints do not have their own pubkey since we don't use it in the protocol yet (this was reserved for future use cases). We currently use the mint URL as an identifier of the mint. Maybe that's a way?